ddl/partition.go

```diff
@@ -457,20 +457,27 @@ func isRangePartitionColUnsignedBigint(cols []*table.Column, pi *model.Partition
 	return false
 }
 
-// truncateTableByReassignPartitionIDs reassign a new partition ids.
-func truncateTableByReassignPartitionIDs(job *model.Job, t *meta.Meta, tblInfo *model.TableInfo) error {
+// truncateTableByReassignPartitionIDs reassigns new partition ids.
+func truncateTableByReassignPartitionIDs(t *meta.Meta, tblInfo *model.TableInfo) error {
 	newDefs := make([]model.PartitionDefinition, 0, len(tblInfo.Partition.Definitions))
 	for _, def := range tblInfo.Partition.Definitions {
 		pid, err := t.GenGlobalID()
 		if err != nil {
-			job.State = model.JobStateCancelled
```

Comment:
Will the caller set it on error ?

Reply:
If only one error occurs, no need to cancel this job, we can retry it later.